### PR TITLE
Changes to travis test. Added stubs folder. Changes to travis.js.

### PR DESCRIPTION
**Pull request by eeswr**

### CODE REVIEW

File: services/travis/lib/travis.js [hunk 1]
```diff
@@ -2,68 +2,72 @@
...
-      var data = null
-
-      try {
-        data = JSON.parse(body)
-      }
-      catch (e) {
-        return done(null, {ok: false, err: e})
-      }
+    catch (e) {
+      return done(err)
```
Comment (mihaidma) [inline services/travis/lib/travis.js]: return done(e)

File: services/travis/srv/start.js [hunk 2]
```diff
@@ -1,49 +1,26 @@
 'use strict'
 
-var Seneca = require('seneca')
-var Entities = require('seneca-entity')
-var Mesh = require('seneca-mesh')
+var Mu = require('mu')
+var Tcp = require('mu/drivers/tcp')
 var Travis = require('../lib/travis')
-var RedisStore = require('seneca-redis-store')
+var Boot = require('boot-in-the-arse')
```
Comment (mihaidma) [inline services/travis/srv/start.js:6]: this was removed. please update like in the latest version

File: services/travis/test/index.test.js [hunk 3]
```diff
@@ -1,183 +1,65 @@
...
+  var mu = Mu()
 
-  it('test ID exists', function (done) {
-    var si = createInstance()
-    var payload = { 'name': 'seneca', 'user': 'powerbrian', 'repo': 'https://github.com/senecajs/seneca' }
-    si.act(_.extend({ role: 'travis', cmd: 'get' }, payload), function (err, reply) {
-      expect(err).to.not.exist()
+  Travis(mu, {}, () => {
+    mu.dispatch({role: 'store', cmd: 'get', type: 'npm', name: 'mu'}, (err, reply) => {
+      var cachedOne = reply.cached
```
Comment (mihaidma) [inline services/travis/test/index.test.js:31]: add also a check `t.ok(cachedOne)`

**Review by mihaidma — CHANGES_REQUESTED** [review]
added some comments, probably the final code will need some tunning but mostly it is ok

File: services/travis/lib/travis.js [hunk 4]
```diff
@@ -2,68 +2,72 @@
...
+module.exports = function (mu, options, done) {
+    opts = Object.assign({}, opts, options)
+    opts.mu = mu
+    opts.cache = Cache(opts.cacheSize)
 
-  opts = seneca.util.deepextend(opts)
-  opts.client = new Travis({version: '2.0.0'})
-  opts.cache = seneca.make$('travis_cache')
-  opts.cache.load$()
+    mu.define({role: 'store', cmd: 'get', type: 'npm'}, cmdGet)
```
Comment (mcdonnelldean) [inline services/travis/lib/travis.js]: Needs to be `tavis` for type

File: services/travis/srv/start.js [hunk 5]
```diff
@@ -1,49 +1,20 @@
...
-}
-else {
-  Service.use(Mesh, opts.mesh)
-  Service.use(RedisStore, opts.redis)
-}
-
-Service.use(Travis, opts.travis)
+var mu = Mu(opts.mu)
+Travis(mu, opts.npm, () => {
+  mu.inbound({role: 'store', type: 'travis'}, Tcp.server({port: '6050'}))
```
Comment (mcdonnelldean) [inline services/travis/srv/start.js]: Change port number to `6052`

**Review by mcdonnelldean — APPROVED** [review]

**Review by mcdonnelldean — APPROVED** [review]

**Review by mihaidma — APPROVED** [review]

**Review by mcdonnelldean — APPROVED** [review]

**Review by mcdonnelldean — APPROVED** [review]

**Pull request merged**